//
// Copyright 2019 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

import "asylo/crypto/certificate.proto";
import "asylo/identity/identity_acl.proto";

// Configuration for an SGX AGE remote assertion authority.
message SgxAgeRemoteAssertionAuthorityConfig {
  // Root CA certificates, besides the Intel root, that are used by this
  // authority.
  repeated Certificate root_ca_certificates = 1;

  // The address and port of the Assertion Generator Enclave service. Required.
  optional string server_address = 2;

  // The identity expectation of the Assertion Generator Enclave. Required.
  optional IdentityAclPredicate age_identity_expectation = 3;

  // The Intel root certificate. Required.
  optional Certificate intel_root_certificate = 4;
}
